recipes/default.rb

```diff
@@ -10,26 +10,14 @@
   compile_time false if respond_to?(:compile_time)
 end
 
-if platform_family? 'rhel'
-  yum_repository 'influxdb' do
-    description 'InfluxDB Repository - RHEL \$releasever'
-    baseurl node['influxdb']['upstream_repository']
-    gpgkey 'https://repos.influxdata.com/influxdb.key'
+influxdb_install 'influxdb' do
+  if node['kernel']['machine'].include? '64'
+    arch_type 'amd64'
+  elsif node['kernel']['machine'].include? '386'
+    arch_type 'i386'
+  else
+    Chef::Log.warning "I am unable to determine arch type for: #{node['kernel']['machine']}"
   end
```

Comment:
@bdangit I think we should handle it a bit more intelligently. I was attempting to mimic current usage (which was hard coded in the recipe only for apt_repository) and assumed there was a reason for this. I can look into this further.

Maybe we could compromise and have a helper function that we call and if we have not specified we can attempt an auto discovery.